onyo/commands/edit.py

```diff
@@ -4,40 +4,43 @@
 import os
 import sys
 
-from git import Repo
+from pathlib import Path
 
 from onyo.utils import (
     build_git_add_cmd,
-    run_cmd,
-    edit_file
+    edit_file,
+    get_editor,
+    get_list_of_assets,
+    run_cmd
 )
 
 logging.basicConfig()
 logger = logging.getLogger('onyo')
 
 
 def build_commit_cmd(files, onyo_root):
-    return ["git -C \"" + onyo_root + "\" commit -m", "edit asset(s).\n\n" + "\n".join(files)]
+    return [f'git -C "{onyo_root}" commit -m "edit asset(s).\n\n"', "\n".join(files)]
 
 
 def prepare_arguments(sources, onyo_root):
     problem_str = ""
     list_of_sources = []
+    asset_list = get_list_of_assets(onyo_root)
+
     if isinstance(sources, str):
         sources = ["".join(sources)]
     for source in sources:
-        current_source = source
-        if not os.path.isfile(current_source):
-            current_source = os.path.join(onyo_root, source)
-        if not os.path.exists(current_source):
-            problem_str = problem_str + "\n" + current_source + " does not exist."
+        current_source = Path(source)
+        if not current_source.is_file():
+            current_source = Path(onyo_root, source)
+        if not current_source.exists():
+            problem_str += f"\n{current_source} does not exist."
         # check if file is in git
-        run_output = run_cmd("git -C \"" + onyo_root + "\" ls-tree -r HEAD ")
-        if source not in run_output:
-            problem_str = problem_str + "\n" + current_source + " is not in onyo."
+        elif True not in [source in asset[0] for asset in asset_list]:
```

Comment:
`True not in []` is more complicated than `False in []`. But Python's truthiness is your friend. You can simplify this to `not []`. For example:

```
elif not [asset[0] for asset in asset_list if source in asset[0]]
```

Reply:
I think `False in []` equals not `not []`:
```
>>> False in [False, False, True]
True
>>> not [False, False, True]
False
```

So I will for now simplify to your proposed  `False in []`, thanks!

Reply:
> I think False in [] equals not not []:

That's why I suggested it. ;-P

> Further update, not any([source in asset[0] for asset in asset_list]) seems to work best and is also fairly readable.

Have you profiled that? I'm not sure if that is faster, as it will need to assign a lot of True/False values and then check them.

Profiling...:
```
def straight(source, asset_list):
    if not [asset[0] for asset in asset_list if source in asset[0]]:
        yope='yope'

# my mispronunciation of "any"
def ernie(source, asset_list):
    if not any([source in asset[0] for asset in asset_list]):
        yope='yope'

asset_list = [(f"{x}", f"{x}") for x in range(1,1000) ]
```
```
In [236]: timeit.timeit(lambda: straight('1', asset_list), number=50000)
Out[236]: 1.3253654120489955

In [237]: timeit.timeit(lambda: straight('no_match', asset_list), number=50000)
Out[237]: 1.0206323569873348

In [238]: timeit.timeit(lambda: ernie('1', asset_list), number=50000)
Out[238]: 1.4346348019316792

In [239]: timeit.timeit(lambda: ernie('no_match', asset_list), number=50000)
Out[239]: 1.5336767170811072
```

Reply:
I did indeed miss-read this, and will update accordingly, thanks.